examples/tokens/ERC1155ownable.vy

```diff
@@ -0,0 +1,356 @@
+        if i >= len(ids):
+            break
+        self.balanceOf[msg.sender][ids[i]] -= amounts[i]
+    
+    log TransferBatch(msg.sender, msg.sender, ZERO_ADDRESS, ids, amounts)
+
+## approval ##
+@external
+def setApprovalForAll(owner: address, operator: address, approved: bool):
+    # @dev set an operator for a certain NFT owner address
+    # @param account the NFT owner address
+    # @param operator the operator address
+    assert owner == msg.sender, "You can only set operators for your own account"
+    assert not self.paused, "The contract has been paused"
+    assert owner != operator, "ERC1155: setting approval status for self"
+    self.isApprovedForAll[owner][operator] = approved
+    log ApprovalForAll(owner, operator, approved)
+
+@external
+def safeTransferFrom(sender: address, receiver: address, id: uint256, amount: uint256, bytes: bytes32):
+    # @dev transfer token from one address to another.
+    # @param sender the sending account (current owner)
+    # @param receiver the receiving account
+    # @param id the token id that will be sent
+    # @param amount the amount of tokens for the specified id
+    assert not self.paused, "The contract has been paused"
+    assert receiver != ZERO_ADDRESS, "ERC1155: transfer to the zero address"
+    assert sender != receiver
+    assert sender == msg.sender or self.isApprovedForAll[sender][msg.sender], "Caller is neither owner nor approved operator for this ID"
+    assert self.balanceOf[sender][id] > 0 , "caller does not own this ID or ZERO balance"
+    operator: address = msg.sender
+    self.balanceOf[sender][id] -= amount
+    self.balanceOf[receiver][id] += amount
+    log TransferSingle(operator, sender, receiver, id, amount)
+
+@external
+def safeBatchTransferFrom(sender: address, receiver: address, ids: DynArray[uint256, BATCH_SIZE], amounts: DynArray[uint256, BATCH_SIZE], _bytes: bytes32):
+    # @dev transfer tokens from one address to another.
+    # @param sender the sending account
+    # @param receiver the receiving account
+    # @param ids a dynamic array of the token ids that will be sent
+    # @param amounts a dynamic array of the amounts for the specified list of ids.
+    assert not self.paused, "The contract has been paused"
+    assert receiver != ZERO_ADDRESS, "ERC1155: transfer to the zero address"
+    assert sender != receiver
+    assert sender == msg.sender or self.isApprovedForAll[sender][msg.sender], "Caller is neither owner nor approved operator for this ID"
+    assert len(ids) == len(amounts), "ERC1155: ids and amounts length mismatch"
+    operator: address = msg.sender
+    for i in range(BATCH_SIZE):
+        if i >= len(ids):
+            break
+        id: uint256 = ids[i]
+        amount: uint256 = amounts[i]
+        self.balanceOf[sender][id] -= amount
+        self.balanceOf[receiver][id] += amount
+    
+    log TransferBatch(operator, sender, receiver, ids, amounts)
+
+# URI #
+@external
+def setURI(uri: String[MAX_URI_LENGTH]):
```

Comment:
should probably check that this is only the owner performing this operation

Reply:
Fixed